### ISSUE TITLE
CORS configuration

### ISSUE THREAD

**Issue by wkulczi** [jaredpetersen/raspi-live#56]
First off, thanks for creating raspilive, it saved me a lot of time setting similar solution.
I have a bit of a problem with connecting served file with web app I've created because of a CORS error. I guess I need to set Access-Control-Allow-Origin to my website address. Can you point me to the direction where should I change the server configuration to edit its CORS headers?
I have zero GOlang experience but I'm willing to learn anything  

Thanks a lot.

**Comment by jaredpetersen** [comment]
Yeah, so it looks like you'd need to add some middleware here to set up your CORs stuff. This should _probably_ be handled by the application out of the box. But in the mean time, I think you need

A new function [here](https://github.com/jaredpetersen/raspilive/blob/463ba8b6ef4627bcaaecf6159d5978d5ddd53aa4/internal/server/static.go#L103):
```go
func corsMiddleware(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		// TODO see https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/http/cors.go#L29-L39
		next.ServeHTTP(w, r)
	})
}
```

And then add it to the middlware chain [here](https://github.com/jaredpetersen/raspilive/blob/463ba8b6ef4627bcaaecf6159d5978d5ddd53aa4/internal/server/static.go#L81)
```
middlewareChain = middlewareChain.Append(corsMiddleware)
```

There's also [this library](https://github.com/rs/cors) which appears to be a little more turn-key but I haven't used it before.

I'll leave this open to remind myself to look more into providing better support for this in the future.

**Comment by wkulczi** [comment]
Can confirm that this solution works, thanks again.

**Comment by matthiasbaldi** [comment]
@jaredpetersen When do you think you will implement it in the raspilive?

**Comment by jaredpetersen** [comment]
Eventually 🤷‍♂️ 

I've been taking a small break from open source to catch up on other hobbies. Is there an urgent need?

**Comment by matthiasbaldi** [comment]
Oh sure, no I think it is also solvable with a reverse proxy in front of the stream.
So it is not urgent ;) take your time. Sadly I can not help with it because it is now written in GO,
I do not have enough knowledge about that ^^
